data/rat.lean

```diff
@@ -211,6 +211,17 @@ begin
     c.dvd_of_dvd_mul_right _),
   have := congr_arg int.nat_abs e,
   simp [int.nat_abs_mul, int.nat_abs_of_nat] at this, simp [this]
+end-/
+
+theorem rat.num_dvd (a) {b : ℤ} (b0 : b ≠ 0) : (a /. b).num ∣ a :=
+begin
+  cases e : a /. b with n d h c,
+  rw [rat.num_denom', rat.mk_eq b0
+    (ne_of_gt (int.coe_nat_pos.2 h))] at e,
+  refine (int.nat_abs_dvd.1 $ int.dvd_nat_abs.1 $ int.coe_nat_dvd.2 $
+    c.dvd_of_dvd_mul_right _),
+  have := congr_arg int.nat_abs e,
+  simp [int.nat_abs_mul, int.nat_abs_of_nat] at this, simp [this]
```

Comment:
What's going on here? It looks like you just commented out and recopied the same theorem

Reply:
Well, I forgot the comment that explained what was going on here (also why I left the old proof there for now). It's not the same theorem: check the type of `b` in the old one. The `b0` argument forces `b` to be a nat, then it gets cast to an int in the type. I'm guessing that's not what you intended, and changing it doesn't break anything, but I meant to ask.

Reply:
oh, I see, it's that stupid nat default thing because of the `0` numeral in `b0`. Yeah, that was not intended. Is there a reason you changed `num_dvd` to `rat.num_dvd` though? Surely we are already in the `rat` namespace, so that would cause double namespacing.

Reply:
Nope, that's another c/p mistake.